addons/website/models/website_menu.py

```diff
@@ -180,10 +181,18 @@ def replace_id(old_id, new_id):
         for menu in data['data']:
             menu_id = self.browse(menu['id'])
             # if the url match a website.page, set the m2o relation
-            # except if the menu url is '#', meaning it will be used as a menu container, most likely for a dropdown
-            if menu['url'] == '#':
+            # except if the menu url has '#', in that case unset the page_id, see below
+            if '#' in menu['url']:
+                # Multiple case possible
+                # 1. # => could be a menu container, most likely for a dropdown
+                # 2. #anchor => append the page from which the menu is saved
+                # 3. /url#something => do nothing as this is a valid URL (current page or not)
+                # 4. https://google.com#something => do nothing, valid external URL
                 if menu_id.page_id:
                     menu_id.page_id = None
+                if menu['url'].startswith('#') and len(menu['url']) > 1 and request:
+                    referer_url = werkzeug.urls.url_parse(request.httprequest.headers.get('Referer')).path
```

Comment:
This could crash if the referrer is hidden
```suggestion
                    referer_url = werkzeug.urls.url_parse(request.httprequest.headers.get('Referer', '')).path
```